Applications/Viqui/vqCore.cxx

```diff
@@ -3387,6 +3393,37 @@ void vqCore::generateReport(QString path, bool generateVideo)
   this->exportKml(path);
 }
 
+//-----------------------------------------------------------------------------
+void vqCore::exportSvm(QString path)
+{
+  QFile file(path);
+  if (!file.open(QIODevice::WriteOnly |
+                 QIODevice::Truncate))
+    {
+    qDebug() << file.errorString();
+    QMessageBox::warning(0, QString(), "Unable to write SVM file.");
+    return;
+    }
+
+  if (this->SavedQueryPlan.isSimilarityQuery())
+    {
+    auto& query = *this->SavedQueryPlan.constSimilarityQuery();
+
+    const int iqrModelSize = static_cast<int>(query.IqrModel.size());
+    if (iqrModelSize)
+      {
+      const char* rawModel =
+        reinterpret_cast<const char*>(&query.IqrModel[0]);
+      const QByteArray encodedModel =
+        QByteArray::fromRawData(rawModel, iqrModelSize).toBase64();
```

Comment:
It might be better to wait to call `toBase64` until the next line (i.e. inside the call to `write`). That said, given that this file contains *only* the IQR model, is there still value in encoding it at all? It seems to me that, here, it might be better to just write it as raw data. (It is encoded for query plans mainly because KST is not intended to hold unencoded raw binary data, much as you would encode a raw PNG/JPG/etc. image when it appears as inline data in XML.)

Reply:
Oops, `toBase64` shouldn't be there at all. This was copy-pasted from `vvXmlUtil`, and I forgot to take that out. Good catch.